lib/board.rb

```diff
@@ -0,0 +1,50 @@
+class Board
+	attr_accessor :cells
+	
+	def initialize
+		@cells = Array.new(9," ")
+	end
+	
+	def reset!
+		@cells = @cells.clear
+		@cells = Array.new(9, " ")
+	end
+	
+	def display
+		puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} \n\
+-----------\n\
+ #{@cells[3]} | #{@cells[4]} | #{@cells[5]} \n\
+-----------\n\
+ #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+	 end
+	
+	def position(pos)
+		@cells[pos.to_i-1]
+	end
+	
+	def full?
+		!@cells.include?(" ")
+	end
+	
+	def turn_count
+		output = 0
+		@cells.each { |x| output+=1 if x!= " "}
+		output
+	end

```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`#count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  @cells.count { |x| x != " " }
end
```